File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.cs
using System;

public class Solution {
    public int[] solution(int[] arr, int[,] queries) {
        int[] answer = new int[queries.GetLength(0)];
        for (int i=0; i<queries.GetLength(0); i++) {
            int s = queries[i,0];
            int e = queries[i,1];
            int k = queries[i,2];
            int save = -1;
            
            for (int j=s; j<=e; j++) {
                if (arr[j] > k) {
                    if (save == -1) {
                        save = arr[j];
                    }
                    else if (save > arr[j]) {
                        save = arr[j];
                    }
                }
            }
        answer[i] = save;
        }
    return answer;
    }
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr, vector<vector<int>> queries) 
{
    vector<int> result;
    
    int row = queries.size();
    int col = queries[0].size();
    
    int s = 0;
    int e = 0;
    int k = 0;
    
    
    vector<int> temp;
    for(int i=0; i<row; i++)
    {
        s = queries[i][0];
        e = queries[i][1];
        k = queries[i][2];
        for(int j=s; j<=e; j++)
        {
            if(arr[j]>k)
            {
                temp.push_back(arr[j]);
            }
        }
        
        if (!temp.empty())
        {
            result.push_back(*min_element(temp.begin(), temp.end()));
        }
        else
        {
            result.push_back(-1);
        }
        temp.clear();
    }
    return result;
}